api/python/t4/packages.py

```diff
@@ -328,7 +323,7 @@ def install(cls, name, registry=None, pkg_hash=None, dest=None, dest_registry=No
         """
         if registry is None:
             registry = get_remote_registry()
-            if not registry:
+            if registry is None:
```

Comment:
why the switch from `not` to `is none`?

Reply:
Well, `if not registry` will fire for anything that evaluates to falsey. `registry is None` will only fire if `registry` is truly `None`. Since we use `None` as our sentinel value (e.g. it's the default value when the user doesn't specify any other value), this is a more semantically correct way of performing a nullity check.

This matters if the user specifies e.g. the following:

```
def install(..., registry="")
```

While this isn't a _correct_ thing to do, it's more correct error behavior to return an error stating that '`""` is a not a valid registry argument' then to discard/ignore this argument and proceed to running the default-registry-setting code.